### PR TITLE
ENH: Logging for skypy command line script and Pipeline class

### PR DESCRIPTION
**Pull request by rrjbca**
## Description
- Setup logger, stream handler and formatter for skypy command line script
- Set the logging level using a command line argument
- Log any exceptions raised while executing the command line script
- Setup a logger for the skypy.pipeline module
- Log each job and function call in the Pipeline class and when writing the output file

Example log output:

```bash
>>> skypy -v examples/galaxies/sdss_photometry.yml galaxies.fits
2021-03-22 19:48:35,758 [INFO] skypy.pipeline: Setting cosmology
2021-03-22 19:48:35,758 [INFO] skypy.pipeline: Calling get
2021-03-22 19:48:35,758 [INFO] skypy.pipeline: Generating red_galaxies
2021-03-22 19:48:35,758 [INFO] skypy.pipeline: Calling Table
2021-03-22 19:48:35,758 [INFO] skypy.pipeline: Generating blue_galaxies
2021-03-22 19:48:35,758 [INFO] skypy.pipeline: Calling Table
2021-03-22 19:48:35,759 [INFO] skypy.pipeline: Generating sky_area
2021-03-22 19:48:35,759 [INFO] skypy.pipeline: Generating magnitude_limit
2021-03-22 19:48:35,759 [INFO] skypy.pipeline: Generating phi_star_red
2021-03-22 19:48:35,759 [INFO] skypy.pipeline: Calling Exponential1D
2021-03-22 19:48:35,759 [INFO] skypy.pipeline: Generating M_star_red
2021-03-22 19:48:35,759 [INFO] skypy.pipeline: Calling Linear1D
2021-03-22 19:48:35,760 [INFO] skypy.pipeline: Generating phi_star
2021-03-22 19:48:35,760 [INFO] skypy.pipeline: Calling Exponential1D
2021-03-22 19:48:35,760 [INFO] skypy.pipeline: Generating M_star
2021-03-22 19:48:35,760 [INFO] skypy.pipeline: Calling Linear1D
2021-03-22 19:48:35,761 [INFO] skypy.pipeline: Generating z_range
2021-03-22 19:48:35,761 [INFO] skypy.pipeline: Calling linspace
2021-03-22 19:48:35,761 [INFO] skypy.pipeline: Generating red_galaxies.redshift, magnitude
2021-03-22 19:48:35,761 [INFO] skypy.pipeline: Calling schechter_lf
2021-03-22 19:48:36,776 [INFO] skypy.pipeline: Generating red_galaxies.spectral_coefficients
2021-03-22 19:48:36,776 [INFO] skypy.pipeline: Calling dirichlet_coefficients
2021-03-22 19:48:36,824 [INFO] skypy.pipeline: Generating red_galaxies.stellar_mass
2021-03-22 19:48:36,824 [INFO] skypy.pipeline: Calling stellar_mass
2021-03-22 19:48:36,848 [INFO] skypy.pipeline: Generating red_galaxies.mag_u, mag_g, mag_r, mag_i, mag_z
2021-03-22 19:48:36,848 [INFO] skypy.pipeline: Calling apparent_magnitudes
2021-03-22 19:48:45,088 [INFO] skypy.pipeline: Generating blue_galaxies.redshift, magnitude
2021-03-22 19:48:45,088 [INFO] skypy.pipeline: Calling schechter_lf
2021-03-22 19:48:47,656 [INFO] skypy.pipeline: Generating blue_galaxies.spectral_coefficients
2021-03-22 19:48:47,656 [INFO] skypy.pipeline: Calling dirichlet_coefficients
2021-03-22 19:48:47,781 [INFO] skypy.pipeline: Generating blue_galaxies.stellar_mass
2021-03-22 19:48:47,781 [INFO] skypy.pipeline: Calling stellar_mass
2021-03-22 19:48:47,793 [INFO] skypy.pipeline: Generating blue_galaxies.mag_u, mag_g, mag_r, mag_i, mag_z
2021-03-22 19:48:47,793 [INFO] skypy.pipeline: Calling apparent_magnitudes
2021-03-22 19:48:57,491 [INFO] skypy: Writing galaxies.fits
```

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/main/CONTRIBUTING.rst)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

File: skypy/pipeline/_pipeline.py [hunk 1]
```diff
@@ -168,6 +169,7 @@ def execute(self, parameters={}):
 
         # go through the jobs in dependency order
         for job in networkx.topological_sort(self.dag):
+            log.info(f"{job}")
```
Comment (ntessore) [inline skypy/pipeline/_pipeline.py:172]: should probably log after `if skip: continue`

File: skypy/pipeline/scripts/skypy.py [hunk 2]
```diff
@@ -14,21 +15,32 @@ def main(args=None):
     parser.add_argument('output', help='Output file name')
     parser.add_argument('-o', '--overwrite', action='store_true',
                         help='Whether to overwrite existing files')
+    parser.add_argument('-l', '--logging', default='NOTSET', help='logging level',
+                        choices=[n for n in logging._nameToLevel if n != 'NOTSET'])
```
Comment (ntessore) [inline skypy/pipeline/scripts/skypy.py:19]: not sure I would waste the short `-l` option on logging

**Review by ntessore — COMMENTED** [review]
I generally prefer repeatable `-v,--verbose` and `-q,--quiet` options but that is a matter of personal taste.

File: docs/pipeline/index.rst [hunk 3]
```diff
@@ -21,6 +21,9 @@ fits files:
 
     $ skypy examples/galaxies/sdss_photometry.yml sdss_photometry.fits
 
+To view the progress of the pipeline as it runs you can enable logging using the
+`--logging INFO` argument.
```
Comment (rrjbca) [inline docs/pipeline/index.rst:25]:
```suggestion
`--verbose` flag.
```

**Review by ntessore — APPROVED** [review]

**Pull request merged**